### DISCUSSION TITLE
Changed/unexpected behavior with pagebreaks in custom headers and heading queries

### DISCUSSION THREAD

**Discussion by lumpiluk** [typst/typst#4839] [Q&A] [answered]
I'm trying to achieve two things: 1) Have chapter headings automatically start on a new page, and 2) have custom page headers that are not shown on chapter title pages.

I tried to create a minimal example, which currently behaves differently in the online and the git version: https://typst.app/project/rALHvQLw0tFFovlB41jGHf

For 2), I have been following [this solution by Yohann](https://stackoverflow.com/questions/76363935/typst-header-that-changes-from-page-to-page-based-on-state/76408469#76408469) on Stackoverflow and recently updated it to use context instead of locate. It involves the following query:

```typst
#let l1-heading = query(
    heading.where(level: 1),
).find(h => h.location().page() == here().page())
```

For 1), I'm using a show rule for level-1 headings that inserts a pagebreak:
```typst
#show heading.where(level: 1): it => [
  #pagebreak()
  Chapter #context counter(heading).display() \
  #it.body
]
```

Now, in the web version the query in my custom header correctly detects whether a level-1 heading is present on a page, despite the pagebreak. In the current git version this does not work anymore, and I'm having a hard time finding a good workaround.

So my questions are: Is the new behavior regarding queries and pagebreaks within headings intended? And if so, what can I do to restore how my custom headers used to work?

**Comment by laurmaedje** [comment, marked as answer]
> So my questions are: Is the new behavior regarding queries and pagebreaks within headings intended? And if so, what can I do to restore how my custom headers used to work?

No, it's a regression and will be fixed very soon.

**Reply by lumpiluk** [reply]
Thanks for the super quick reply!

**Reply by laurmaedje** [reply]
Here's the fix by the way: https://github.com/typst/typst/pull/4840

Coincidentally, this was what I was just working on.
